app/market/item/[id]/page.jsx

```diff
@@ -1,43 +1,136 @@
-
-/**
- * @param {params} Object
- 
-*/
-export default function Page({ params }) {
-  const [data, setData] = useState(null);
+import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
+import { theme } from "@/styles/theme";
+
+// Function to convert currency code to symbol
+const getCurrencySymbol = (currencyCode) => {
+  const currencySymbols = {
+    USD: "$",
+    EUR: "€",
+    GBP: "£",
+  };
+  return currencySymbols[currencyCode] || currencyCode;
+};
+
+const IndividualCardPage = ({ params }) => {
+  const [cardDetails, setCardDetails] = React.useState(null);
   const router = useRouter();
-  const id = params.id;
-  useEffect(() => {
-    //fetch data need to move to file in utils
-    const fetchCard = async () => {
-      const response = await fetch(`/api/cards/${id}`);
-      if (!response.ok) {
-        throw new Error(response.statusText);
+  const { id } = params;
+
+  React.useEffect(() => {
+    const fetchCardDetails = async () => {
+      try {
+        const response = await fetch(`/api/cards/${id}`);
+        if (!response.ok) {
+          throw new Error('Failed to fetch card details');
+        }
+        const data = await response.json();
+        setCardDetails(data.data); // Update to set the entire data object
+      } catch (error) {
+        console.error('Error fetching card details:', error);
+        setCardDetails(null);
       }
-      const data = await response.json();
-      setData(data.data);
     };
-    fetchCard();
+
+    if (id) {
+      fetchCardDetails();
+    }
   }, [id]);
 
+  const handleEditButtonClick = () => {
+    router.push(`/sell/edit/${id}`);
+  };
+
+  const handleSellerInfoButtonClick = () => {
+    router.push(`/market/seller/${id}`);
+  };
+
+  const handleAddToCartButtonClick = () => {
+    router.push("/cart");
+  };
+
   return (
-    <Box>
-      {data && <CardComponent card={data} />}
-      <Button variant="contained" color="primary" onClick={() => router.push(`/sell/edit/${id}`)}>
-        Edit card
-      </Button>
-      <Button variant="contained" color="primary" onClick={() => router.push(`/market/seller/[id]`)}>
-        Watch information about the seller
-      </Button>
-      <Button variant="contained" color="primary" onClick={() => router.push("/cart")}>
-        Add card to cart
-      </Button>
-    </Box>
+    <ThemeProvider theme={theme}>
+      <Box style={{ marginLeft: theme.spacing(2) }}>
+        {/* Breadcrumbs */}
+        <Breadcrumbs aria-label="breadcrumb" style={{ marginTop: '8px' }}>
+          <Link color="inherit" href="/" onClick={() => router.push('/')}>
+            Home
+          </Link>
+          <Typography color="text.primary">Card Details</Typography>
+        </Breadcrumbs>
+
+        {/* Image and Details Section */}
+        <div style={{ display: 'flex', marginTop: theme.spacing(2)}}>
+          {/* Image Section */}
+          {cardDetails && (
+            <Card style={{ boxShadow: 'none', marginRight: theme.spacing(2) }}>
```

Comment:
we should use 
our CardComponent. its height will be fixed in the Luba's pull request.